code/modules/admin/verbs/adminpm.dm

```diff
@@ -65,131 +98,115 @@
 		return
 
 	var/client/C
-	var/irc = 0
 	if(istext(whom))
 		if(cmptext(copytext(whom,1,2),"@"))
 			whom = findStealthKey(whom)
-		if(whom == "IRCKEY")
-			irc = 1
-		else
-			C = directory[whom]
+		C = directory[whom]
 	else if(istype(whom,/client))
 		C = whom
-	if(irc)
-		if(!ircreplyamount)	//to prevent people from spamming irc
-			return
-		if(!msg)
-			msg = input(src,"Message:", "Private message to Administrator") as text|null
-
-		if(!msg)
-			return
-		if(holder)
-			src << "<font color='red'>Error: Use the admin IRC channel, nerd.</font>"
-			return
+	if(!C)
+		if(holder)	src << "<font color='red'>Error: Admin-PM: Client not found.</font>"
+		else		adminhelp(msg)	//admin we are replying to left. adminhelp instead
+		return
 
+	//get message text, limit it's length.and clean/escape html
+	if(!msg)
+		msg = input(src,"Message:", "Private message to [key_name(C, 0, 0)]") as text|null
 
-	else
+		if(!msg)	return
 		if(!C)
-			if(holder)
-				src << "<font color='red'>Error: Admin-PM: Client not found.</font>"
-			else
-				adminhelp(msg)	//admin we are replying to left. adminhelp instead
+			if(holder)	src << "<font color='red'>Error: Admin-PM: Client not found.</font>"
+			else		adminhelp(msg)	//admin we are replying to has vanished, adminhelp instead
 			return
 
-		//get message text, limit it's length.and clean/escape html
-		if(!msg)
-			msg = input(src,"Message:", "Private message to [key_name(C, 0, 0)]") as text|null
-
-			if(!msg)
-				return
-
-			if(prefs.muted & MUTE_ADMINHELP)
-				src << "<font color='red'>Error: Admin-PM: You are unable to use admin PM-s (muted).</font>"
-				return
-
-			if(!C)
-				if(holder)
-					src << "<font color='red'>Error: Admin-PM: Client not found.</font>"
-				else
-					adminhelp(msg)	//admin we are replying to has vanished, adminhelp instead
-				return
-
 	if (src.handle_spam_prevention(msg,MUTE_ADMINHELP))
 		return
 
 	//clean the message if it's not sent by a high-rank admin
-	if(!check_rights(R_SERVER|R_DEBUG,0)||irc)//no sending html to the poor bots
+	if(!check_rights(R_SERVER|R_DEBUG,0))
 		msg = sanitize(copytext(msg,1,MAX_MESSAGE_LEN))
-		if(!msg)
-			return
-
-	var/rawmsg = msg
-
-	if(holder)
-		msg = emoji_parse(msg)
+		if(!msg)	return
 
+	msg = emoji_parse(msg)
 	var/keywordparsedmsg = keywords_lookup(msg)
-
-	if(irc)
-		src << "<font color='blue'>PM to-<b>Admins</b>: [rawmsg]</font>"
-		ircreplyamount--
-		send2irc("Reply: [ckey]",rawmsg)
-		send2admindiscord("Reply: [ckey] [rawmsg]")
-	else
-		if(C.holder)
-			if(holder)	//both are admins
+	var/datum/adminticket/ticket
+
+	if(C.holder)
+		if(holder)	//both are admins
+			for(var/datum/adminticket/T in admintickets)
+				if(T.permckey == src.ckey && T.resolved == "No")
+					T.logs += "<span class='notice'>[src] TO [C]: [msg] </span>"
+					ticket = T

```

Comment:
Will there ever be more than one admin ticket open per client? If not I think we should break here...